src/poll.rs

```diff
@@ -611,7 +613,9 @@ mod syscall {
     #[inline]
     fn cvt(len: isize) -> io::Result<usize> {
         if len == -1 {
```

Comment:
See comments above. `if len < 0 {` is the correct way. All other parts are in my opinion correct.